### DISCUSSION TITLE
Is there a way to prevent it from speaking the things you say?

### DISCUSSION THREAD

**Discussion by OmsagroSylph** [karashiiro/TextToTalk#94] [Q&A] [answered]
I would like to prevent it from reading out things typed by me.  I tried to use an exclusion which didn't seem to work, but maybe I didn't format things correctly.

**Comment by ryankhart** [comment]
@OmsagroSylph Karashiiro created an issue for this request here https://github.com/karashiiro/TextToTalk/discussions/94. That'll be the one page you'll want to subscribe to for notifications on this.

**Comment by ryankhart** [comment, marked as answer]
Oh actually, I just tried out your idea to use an exclusion to prevent it from reading your chat text that you've typed, and I seem to have it working for me. Try typing your first and last name (case-sensitive just in case) with no regex box checked. It should detect your character's name and not read it aloud.

But this feedback is still valid as a feature request to have an option to automatically detect the player's name and filter it out in the chat box.

**Reply by karashiiro** [reply]
That's too broad, isn't it? That will prevent any text from NPCs that uses your full name from being spoken, as well. It's probably more robust to just have a configuration option that checks the sender of chat messages.

**Reply by ryankhart** [reply]
Oh right! I didn't think of that in the moment. I just now tried adding a colon after my character's last name, didn't work. With regex, I tried `First Last\:` `First Last[\:]` `First Last[:]` `First\ Last\:` etc. I couldn't seem to get it to work.

**Reply by ryankhart** [reply]
Unless maybe the filtering only happens after the text gets converted to the output text. I'll try `Firstname Lastname says`

**Reply by karashiiro** [reply]
https://github.com/karashiiro/TextToTalk/blob/main/src/TextToTalk/Middleware/ChatMessageHandler.cs#L61-L85 it does, but I think that's actually a bug 🥴

**Reply by karashiiro** [reply]
https://github.com/karashiiro/TextToTalk/blob/main/src/TextToTalk/Middleware/ChatMessageHandler.cs#L61 I meant checking against this field.

**Reply by ryankhart** [reply]
That worked, but just in case an NPC uses those 3 words in the middle of a sentence, which I suppose is theoretically possible, throw in a `^` character at the beginning and check the regex box so it forces it to match "Firstname Lastname says" but only if it's at the very beginning of the line.

So: `^Firstname Lastname says`

**Reply by OmsagroSylph** [reply]
> That worked, but just in case an NPC uses those 3 words in the middle of a sentence, which I suppose is theoretically possible, throw in a `^` character at the beginning and check the regex box so it forces it to match "Firstname Lastname says" but only if it's at the very beginning of the line.
> 
> So: `^Firstname Lastname says`

This seems to work, I dont know why First Last didn't seem to work for me, but this will do. Thank you.

**Reply by OmsagroSylph** [reply]
> > That worked, but just in case an NPC uses those 3 words in the middle of a sentence, which I suppose is theoretically possible, throw in a `^` character at the beginning and check the regex box so it forces it to match "Firstname Lastname says" but only if it's at the very beginning of the line.
> > So: `^Firstname Lastname says`
> 
> This seems to work, I dont know why First Last didn't seem to work for me, but this will do. Thank you.

Correction, this doesn't work in a party, have to make a second entry for "^.{1}", but also it fails if you use the option to only have it say the name once.
